staff/demi/task-app-complete/task-app/src/index.js

```diff
@@ -0,0 +1,64 @@
+require('dotenv').config()
+
+const express = require('express')
+
+const bodyParser = require('body-parser')
+
+const taskApi = require('./taskapi')
+
+const app = express()
+
+
+app.set('view engine', 'pug')
+
+app.get('/', (req, res) => {
+    
+    dones = todos = []
+
+    taskApi.getTasks().then(tasks => {
+
+        const todos = tasks.filter(task => task.status == false)
+
+        const dones = tasks.filter(task => task.status == true)
+
+        res.render('index', {todos, dones})
+
+
+    })
+
+})
+
+const formBodyParser = bodyParser.urlencoded({ extended: false })
+
+app.post('/tasks', formBodyParser, (req, res) => {
+    const { body: { text } } = req
+
+    taskApi.createTask(text); 
+
+    res.redirect('/')
```

Comment:
Hay que tener en cuenta que estáis usando métodos asíncronos y que, por lo tanto, es posibe que al hacer el redirect, la página llegase a a renderizar sin resolver la 'promise'. 
Como sugerencia, se debería incluir el 'redirect' dentro de un .then(), de manera que no se espera para ejecutar el 'redirect'.

Reply:
I Agree AlexGonRod.